### PR TITLE
Log-normal log-likelihood

### PR DESCRIPTION
**Pull request by ben18785**
Adds a log-normal distribution both without (default) and with mean-adjustment as described in #1376

### CODE REVIEW

File: pints/_log_likelihoods.py [hunk 1]
```diff
@@ -777,6 +777,122 @@ def __init__(self, problem, sigma):
...
+
+        self._mean_adjust = mean_adjust
+
+    def __call__(self, x):
+        sigma = np.asarray(x[-self._no:])
+        if any(sigma < 0):
+            return -np.inf
+
+        soln = self._problem.evaluate(x[:-self._no])
+        if np.any(soln < 0):
```
Comment (DavAug) [inline pints/_log_likelihoods.py:849]: The pythonic way to do this and a to have a try except statement around the error computation below. Because soln <= 0 will lead to an exception in np.log(soln), you can catch it there. 
This also saves a little bit of unnecessary masking.

File: pints/_log_likelihoods.py [hunk 2]
```diff
@@ -777,6 +777,122 @@ def __init__(self, problem, sigma):
...
+
+        sigma = np.asarray(x[-self._no:])
+
+        # Evaluate, and get residuals
+        y, dy = self._problem.evaluateS1(x[:-self._no])
+
+        # Reshape dy, in case we're working with a single-output problem
+        dy = dy.reshape(self._nt, self._no, self._n_parameters - self._no)
+
+        # Note: Must be (np.log(data) - simulation), sign now matters!
```
Comment (DavAug) [inline pints/_log_likelihoods.py:875]: I saw this comment also in other log-likelihoods and I got confused because the sign only matters be because we assume this sign for the partial derivative below 😂 
In general the sign doesn't matter. Could you specify that?

File: pints/tests/test_log_likelihoods.py [hunk 3]
```diff
@@ -1405,6 +1405,150 @@ def simulate(self, x, times):
...
+                          problem)
+
+    def test_call(self):
+        # test calls of log-likelihood
+
+        # single output problem
+        sigma = 1
+        mu = 3.7
+        log_like = self.log_likelihood([mu, sigma])
+        self.assertAlmostEqual(log_like, -10.164703123713256)
```
Comment (DavAug) [inline pints/tests/test_log_likelihoods.py:1448]: I personally don't think that those tests are particularly meaningful, because those are just numbers I can't really interpret. Does it potentially make more sense to test known edge cases to make sure the likelihood's and sensitivities are correct? 
For example the expressions reduce when soln=1, or sigma=1. Then you could compare manually computer values from the parameters, which are easier to understand, and compare them to the methods.

**Review by DavAug — CHANGES_REQUESTED** [review]
Looks really good @ben18785 ! I like the Boolean for switching between the means. I just have a couple minor comments and a general comment on testing.

**Review by DavAug — APPROVED** [review]

**Comment by MichaelClerx** [comment]
@ben18785 @DavAug this is approved. Want to merge it in now?

File: pints/_log_likelihoods.py [hunk 4]
```diff
@@ -777,6 +777,123 @@ def __init__(self, problem, sigma):
...
+    .. math::
+        \log y(t) \sim \text{Normal}(\log f(\theta, t), \sigma)
+
+    Important to note is that:
+
+    .. math::
+        \mathbb{E}(y(t)) = f(\theta, t) \exp(\sigma^2 / 2)
+
+    As such, the optional parameter `mean_adjust` (if true) adjusts the
+    mean of the distribution so that its expectation is ``f(\theta, t)``. This
```
Comment (MichaelClerx) [inline pints/_log_likelihoods.py:802]:
```suggestion
    mean of the distribution so that its expectation is :math:`f(\theta, t)`. This
```
